### DISCUSSION TITLE
At a lost with Synology NAS running Homebridge on Docker

### DISCUSSION THREAD

**Discussion by speciouspresent** [AlexGustafsson/homebridge-wol#165] [Questions]
I am currently running homebridge on my Synology via Docker and have installed `homebridge-wol`. I am a beginner and don't know much about networking. 

I have several questions that I hope someone can help with. 

1. Can someone confirm my understanding of things I need to do to get `wake on lan` and `sleep on lan` to work on my Windows 10 PC? There are as follows:
 - Configure my PC to accept WOL magic packets. 
 - Confirm that wake and sleep commands sent to PC work.
 - Configure `homebridge-wol` by adding the wake command and sleep commands via homebridge UI -> Plugins -> homebridge-wol.

If my understanding is correct then... here are my next questions:

2. How can I confirm that my PC responds to `wake` or `sleep` commands?
- What settings do I need to enable on my PC (running Windows 10)? 
- How do I test the `wake` or `sleep` command? Is there a step by step guide for this? (preferably without installing any programs).

3. After I've confirmed that my PC is responsive to `wake` or `sleep` commands, how do I configure `homebridge-wol`?
- The `readme.md` says that I will need to enable permissions on homebridge. I have read through #153 and don't even know how to create my own image by using `oznu/docker-homebridge` as the base. Is there a more detailed guide I can reference for this?
- What command should I enter via homebridge UI -> Plugins -> homebridge-wol to `wake` or `sleep` the PC?

Really sorry for the questions ! I am just a beginner that is really interested in homebridge and automation. I've managed to survived so far via trial and error, but have hit a brick wall with trying to get this to work.

**Comment by AlexGustafsson** [comment]
Sorry for the late response, did you end up solving it?

1. For wake on lan you do not need the "wake" command. It's only useful if you don't want to send the wake on lan magic packets. You don't strictly need to use any sleep command for the plugin to function, but yes, if you want to turn your computer off or the like, you'll need some sort of command to do so remotely.

2. The easiest way is probably by using some third party wake on lan application, be it a CLI tool or a GUI application. As for testing the wake and shutdown commands, that's easy! On the computer running Homebridge, open up a terminal and simply run the command there. If it works, it will work with this plugin. 

Say for example you want to use `ssh pc shutdown -h now` as the command to shutdown your PC over SSH. Simply run the command in a terminal and see if your computer indeed turns off. If it doesn't, the terminal will give you all the information it can as to why it didn't work.

You don't necessarily need to enable permissions, but you may. Your safest bet is to first try without enabling them. If you get weird errors such as "unable to ping", "permission denied" and so on, you likely need to allow the container to use raw sockets. How to do that, however, varies a lot from platform to platform. It's rather unlikely that you would need to create a custom Docker image, so don't worry about that.

I don't use the UI so unfortunately I don't know if you can run a command to use this plugin. If you're presented with a regular terminal, you can run the commands you've specified in your config yourself, just as you would when testing the commands.

Regarding guides, besides old issues, there is a wiki post that was contributed, available here: https://github.com/AlexGustafsson/homebridge-wol/wiki/Running-on-Synology-Docker-and-turn-Windows-10-on-off-using-SSH-and-Samba. Unfortunately I don't have much time to write new guides - partly because the plugin is really versatile in the sense that there's so many use cases to cover.

If you've been able to solve the issue, don't hesitate to document it and post about it here or in the wiki to help others in the future.

PS. I'm sorry for the late answer, I had missed your question. Hopefully you were able to solve it or found another solution that fit your needs.
